### ISSUE TITLE
SnapshotManager can't handle rollback calls that would pick the current snapshot as target snapshot.

### ISSUE THREAD

**Issue by szlta** [apache/iceberg#5507]
### Apache Iceberg version

0.14.0 (latest release)

### Please describe the bug 🐞

Example test case for `TestSnapshotManager.java`:
```
  @Test
  public void testAttemptToRollbackToCurrentSnapshot() {
    table.newAppend().appendFile(FILE_A).commit();

    long currentSnapshotTimestampPlus100 = table.currentSnapshot().timestampMillis() + 100;
    table
        .manageSnapshots()
        .rollbackToTime(currentSnapshotTimestampPlus100)
        .commit();

    long currentSnapshotId = table.currentSnapshot().snapshotId();
    table
        .manageSnapshots()
        .rollbackTo(currentSnapshotId)
        .commit();
  }
```
The above case fails with 
```
Cannot commit transaction: last operation has not committed
java.lang.IllegalStateException: Cannot commit transaction: last operation has not committed
	at org.apache.iceberg.relocated.com.google.common.base.Preconditions.checkState(Preconditions.java:502)
	at org.apache.iceberg.BaseTransaction.commitTransaction(BaseTransaction.java:252)
	at org.apache.iceberg.SnapshotManager.commit(SnapshotManager.java:158)
	...
```
I think in such cases the API should still return without errors, even if the request turned out to be a no-op.
This was working well in 0.13.0, but since 0.14.0 it is failing due to the refactor in this commit: https://github.com/apache/iceberg/commit/e41e8de673628baa6d641f5ad6b03680c18cc0fe

The reason for the failure is that `SetSnapshotOperation#commit()` doesn't commit on `taskOps` at https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/SetSnapshotOperation.java#L126-L137 leaving  https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/BaseTransaction.java#L105 in false state and eventually causing the transaction commit to throw the above exception.

IMHO there isn't a good reason for `SetSnapshotOperation#commit()` to return without committing in such cases as the commitTransaction() invocation would do a similar logic later anyway.
So I think the solution would be to remove this "return if no changes" logic from `SetSnapshotOperation`. @rdblue - as the original contributor, what's your opinion?
cc: @lcspinter @pvary

**Comment by kbendick** [comment]
Since the above refactoring has a transaction being used, it seems like we need to discard the changes and/or commit the transaction (possibly as a no-op).

This does seem like something that should be allowed to proceed though, unless there's a reason not to allow it.

**Comment by amogh-jahagirdar** [comment]
Yeah I think it makes sense to commit the SetSnapshotOperation. I remember a discussion around this came up in https://github.com/apache/iceberg/pull/4071#discussion_r837008625  for updating snapshot references. We commit the operation even if there are no changes.

Another way I was thinking was in BaseTransaction we https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/BaseTransaction.java#L502 we just advance the "lastOperationCommitted" flag in the case there is no intermediate snapshots or metadata updates. That way we avoid having to do a commit in the first place because it seems as though the original concern is the extra call required for commit?

**Comment by szlta** [comment]
Thanks all for the feedback here. I think [as referenced] since `UpdateSnapshotReferenceOperation` is comitting empty operations in a similar way already, we should make `SetSnapshotOperation` behave the same way. I think the empty commit in such cases will not cause noticeable performance degradation, and other than that I don't see a good reason why to keep that early `return` statement.

I've opened a PR to make it happen at:
https://github.com/apache/iceberg/pull/5536

**Comment by rdblue** [comment]
It seems reasonable to set `lastOperationCommitted` in the path where we skip an actual commit. Should we try that and see if it fixes it?

**Comment by rdblue** [comment]
I just merged PR #5536 to fix this. That fix makes sense to me because we can always avoid swapping identical metadata in `TableOperations` rather than not calling `commit`.
